qcodes/dataset/sqlite/queries.py

```diff
@@ -461,27 +552,65 @@ def get_parameter_tree_values(
         end: The (1-indexed) result to include as the last result to be
             returned. None is equivalent to "all the rest". If start > end,
             nothing is returned.
+        callback: Function called during the data loading every
+            config.dataset.callback_percent.
 
     Returns:
         A list of list. The outer list index is row number, the inner list
         index is parameter value (first toplevel_param, then other_param_names)
     """
 
+    cursor = conn.cursor()
+    offset: int | np.ndarray
+
     offset = max((start - 1), 0) if start is not None else 0
     limit = max((end - offset), 0) if end is not None else -1
 
     if start is not None and end is not None and start > end:
         limit = 0
 
+    # start and end currently not working with callback
+    if start is None and end is None and callback is not None:
+        offset, limit = _get_offset_limit_for_callback(
+            conn, result_table_name, toplevel_param_name
+        )
+
+    # Create the base sql query
     columns = [toplevel_param_name] + list(other_param_names)
     sql = f"""
-    SELECT {','.join(columns)} FROM "{result_table_name}"
-    WHERE {toplevel_param_name} IS NOT NULL
-    LIMIT ? OFFSET ?
-    """
-    cursor = conn.cursor()
-    cursor.execute(sql, (limit, offset))
-    return many_many(cursor, *columns)
+           SELECT {','.join(columns)} FROM "{result_table_name}"
+           WHERE {toplevel_param_name} IS NOT NULL
+           LIMIT ? OFFSET ?
+           """
+
+    # Request if no callback
+    if callback is None:
+
+        cursor.execute(sql, (limit, offset))
+        res = many_many(cursor, *columns)
+
+    # Request if callback
+    elif callback is not None:
+        assert isinstance(offset, np.ndarray)
+
+        # 0
+        progress = 0.0
+        callback(progress)
+
+        # 1
+        cursor.execute(sql, (limit, offset[0]))
+        res = many_many(cursor, *columns)
+        progress += config.dataset.callback_percent
```

Comment:
This does not correctly take into account what happens when there are less than 100 rows

Reply:
Yep, should be fixed